annotations/views/repository_views.py

```diff
@@ -239,18 +240,67 @@ def repository_collection_texts(request, repository_id, group_id, group_collecti
         return render(request, 'annotations/repository_ioerror.html', {'error': str(e)}, status=500)
 
     project_id = request.GET.get('project_id', None)
+    
+    # Pagination
+    page = request.GET.get('page', 1)
+    paginator = Paginator(texts['items'], 20)  # Show 20 texts per page
+    
+    try:
+        paginated_texts = paginator.page(page)
+    except PageNotAnInteger:
+        paginated_texts = paginator.page(1)
+    except EmptyPage:
+        paginated_texts = paginator.page(paginator.num_pages)
+
     context = {
         'user': user,
         'repository': repository,
-        'texts': texts['items'],
+        'texts': paginated_texts,
         'title': f'Texts in Collection: ', # Collection name is rendered from frontend
-        'group_info':texts['group'],
-        'group_id':group_id,
+        'group_info': texts['group'],
+        'group_id': group_id,
         'project_id': project_id,
     }
 
     return render(request, 'annotations/repository_collections_text_list.html', context)
 
+@citesphere_authenticated
+def repository_group_texts(request, repository_id, group_id):
+    """View to fetch and display texts within a group from Citesphere"""
+    user = request.user
+    repository = get_object_or_404(Repository, pk=repository_id)
+    manager = RepositoryManager(user=user, repository=repository)
+
+    try:
+        texts = manager.group_items(group_id)
+    except Exception as e:
```

Comment:
Currently in the manager function this is how an error is being handled:
```
        if response.status_code == 200:
            return response.json()  # Return the groups data
        else:
            response.raise_for_status()
```
If an API response is not 200, it will return a generic HTTPError. Then the subsequent view shows an error page. This is set for all manager functions and its following views. To maintain consistency, can I edit the error handling in [VOGRE-9](https://diging.atlassian.net/browse/VOGRE-9?atlOrigin=eyJpIjoiMDM3MTM0YWE4ZDI1NDQxY2I0NDc2Mjk2ZjE3OGYxMjUiLCJwIjoiaiJ9) since im editing all manager functions, I can change it handle specific errors and show different error messages and apply it all of the functions and views.

Reply:
ya, that's fine